src/akkudoktoreos/core/pydantic.py

```diff
@@ -35,6 +35,10 @@
 from akkudoktoreos.utils.datetimeutil import to_datetime, to_duration
 
 
+class ParametersBaseModel(BaseModel):
+    model_config = ConfigDict(extra="forbid")
+
+
```

Comment:
If this only holds for optimization parameters, please move to optimization/optimizationabc.py and call it OptimizationParametersBase. Why do you use BaseModel? The custom PydanticBaseModel supports several custom types that are not supported by BaseModel.